Crown-Commercial-Service/crown-marketplace
# FM-1266- improving test coverage for the procurements_controller

Before this work was done the coverage was 46.12%
After finishing the work the coverage is now 94.36%

- Add tests for the show pages
- Added tests for updates concerning results
- testing that the determination of state changes is implemented correctly from the results page
- Tests for continue of da
- Fix bug when user goes to procurement url in sent state
- Add tests for the routing on da_draft show pages
- Add tests for the invoice details in the controller
- Add tests for the authorised and notice  details in the controller
- Update methods in controller
- Test for saving new address on contact details pages
- Refactored controller to remove excess code
- Add tests for saving new contact details
- Refactor controller and remove redundant methods
- Add tests for update pension funds
- Update services and regions methods in controller
- Add tests for the rendering of the edit page
- Added tests for creating contact details
- Add tests for removing the contact details of necessary
- Test for building services page
- Remove unused methods and refactor
- Add tests for the destroy method
